org.uqbar.project.wollok/src/org/uqbar/project/wollok/validation/WollokDslValidator.xtend

```diff
@@ -442,40 +493,68 @@ class WollokDslValidator extends AbstractConfigurableDslValidator {
 	@CheckGroup(WollokCheckGroup.POTENTIAL_PROGRAMMING_PROBLEM)
 	def voidMessagesCannotBeUsedAsValues(WMemberFeatureCall call) {
 		val method = call.resolveMethod(classFinder)
-		if (method !== null && !method.native && !method.abstract && !method.supposedToReturnValue && call.isUsedAsValue(classFinder)) {
-			report(WollokDslValidator_VOID_MESSAGES_CANNOT_BE_USED_AS_VALUES, call, WMEMBER_FEATURE_CALL__FEATURE, VOID_MESSAGES_CANNOT_BE_USED_AS_VALUES)
+		if (method !== null && !method.native && !method.abstract && !method.supposedToReturnValue &&
+			call.isUsedAsValue(classFinder)) {
+			report(WollokDslValidator_VOID_MESSAGES_CANNOT_BE_USED_AS_VALUES, call, WMEMBER_FEATURE_CALL__FEATURE,
+				VOID_MESSAGES_CANNOT_BE_USED_AS_VALUES)
 		}
 	}
 
+	// TODO: a single method performs many checks ! cannot configure that
 	@Check
-	//TODO: a single method performs many checks ! cannot configure that
 	@CheckGroup(WollokCheckGroup.POTENTIAL_PROGRAMMING_PROBLEM)
 	def unusedVariables(WVariableDeclaration it) {
 		val assignments = variable.assignments
 		if (assignments.empty) {
 			if (writeable)
-				warning(WollokDslValidator_WARN_VARIABLE_NEVER_ASSIGNED, it, WVARIABLE_DECLARATION__VARIABLE, VARIABLE_NEVER_ASSIGNED)
+				warning(WollokDslValidator_WARN_VARIABLE_NEVER_ASSIGNED, it, WVARIABLE_DECLARATION__VARIABLE,
+					VARIABLE_NEVER_ASSIGNED)
 			else if (!writeable)
-				error(WollokDslValidator_ERROR_VARIABLE_NEVER_ASSIGNED, it, WVARIABLE_DECLARATION__VARIABLE, VARIABLE_NEVER_ASSIGNED)
+				error(WollokDslValidator_ERROR_VARIABLE_NEVER_ASSIGNED, it, WVARIABLE_DECLARATION__VARIABLE,
+					VARIABLE_NEVER_ASSIGNED)
 		}
 		if (variable.uses.empty)
-			warning(WollokDslValidator_VARIABLE_NEVER_USED, it, WVARIABLE_DECLARATION__VARIABLE, WARNING_UNUSED_VARIABLE)
+			warning(WollokDslValidator_VARIABLE_NEVER_USED, it, WVARIABLE_DECLARATION__VARIABLE,
+				WARNING_UNUSED_VARIABLE)
+	}
+
+//  Confirm with @npasserini
+//  @dodain - I think we should not consider this
+//	@Check
+//	@CheckGroup(WollokCheckGroup.POTENTIAL_PROGRAMMING_PROBLEM)
+//	def unusedParameters(WMethodDeclaration it) {
+//		checkUnusedParameters(it.parameters)
+//	}
```

Comment:
Maybe we can check if there is not other method with the same signature. Because if there is not other method the parameter can be removed. If there is other method, I could not know as the parameter can be used by the other object. But I think is quite complicated to check if there is another method. If we don't check that I think we should remove it. Please remove the commented code if @npasserini is ok with it.